pkg/kernels/config.go

```diff
@@ -0,0 +1,164 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Tetragon
+
+//go:build !windows
+
+package kernels
+
+import (
+	"bufio"
+	"compress/gzip"
+	"io"
+	"os"
+	"strings"
+	"sync"
+
+	"golang.org/x/sys/unix"
+)
+
+type Config int
+
+const (
+	CONFIG_AUDIT Config = iota
+	CONFIG_BPF
+	CONFIG_BPF_EVENTS
+	CONFIG_BPF_JIT
+	CONFIG_BPF_JIT_DEFAULT_ON
+	CONFIG_BPF_KPROBE_OVERRIDE
+	CONFIG_BPF_SYSCALL
+	CONFIG_CGROUPS
+	CONFIG_DEBUG_INFO_BTF
+	CONFIG_DEBUG_INFO_BTF_MODULES
+	CONFIG_FTRACE_SYSCALLS
+	CONFIG_HAVE_BPF_JIT
+	CONFIG_HAVE_EBPF_JIT
+	CONFIG_SECURITY
+)
+
+type kernelConfig struct {
+	name  string
+	value string
+}
+
+var (
+	getKernelConfig = sync.OnceValue(initKernelConfig)
+	kernelConfigMap = map[Config]kernelConfig{
+		CONFIG_AUDIT:                  {name: "CONFIG_AUDIT"},
+		CONFIG_BPF:                    {name: "CONFIG_BPF"},
+		CONFIG_BPF_EVENTS:             {name: "CONFIG_BPF_EVENTS"},
+		CONFIG_BPF_JIT:                {name: "CONFIG_BPF_JIT"},
+		CONFIG_BPF_JIT_DEFAULT_ON:     {name: "CONFIG_BPF_JIT_DEFAULT_ON"},
+		CONFIG_BPF_KPROBE_OVERRIDE:    {name: "CONFIG_BPF_KPROBE_OVERRIDE"},
+		CONFIG_BPF_SYSCALL:            {name: "CONFIG_BPF_SYSCALL"},
+		CONFIG_CGROUPS:                {name: "CONFIG_CGROUPS"},
+		CONFIG_DEBUG_INFO_BTF:         {name: "CONFIG_DEBUG_INFO_BTF"},
+		CONFIG_DEBUG_INFO_BTF_MODULES: {name: "CONFIG_DEBUG_INFO_BTF_MODULES"},
+		CONFIG_FTRACE_SYSCALLS:        {name: "CONFIG_FTRACE_SYSCALLS"},
+		CONFIG_HAVE_BPF_JIT:           {name: "CONFIG_HAVE_BPF_JIT"},
+		CONFIG_HAVE_EBPF_JIT:          {name: "CONFIG_HAVE_EBPF_JIT"},
+		CONFIG_SECURITY:               {name: "CONFIG_SECURITY"},
+	}
+)
+
+func setKernelConfigMap(key, value string) {
+	for index, config := range kernelConfigMap {
+		if config.name == key {
+			config.value = value
+			kernelConfigMap[index] = config
+			break
+		}
+	}
+}
```

Comment:
Could you simplify this by having just a `kernelConfigMap = map[string]string` with the key being the name and the value being `m`, `y` or `n`? Not sure what's the benefit of having the extra int enum?